### ISSUE TITLE
Allow responsive values for `variant`, `size`, and `orientation`

### ISSUE THREAD

**Issue by with-heart** [chakra-ui/chakra-ui#1028]
The new modifier versions of the `variant` and `size` props don't accept responsive values (specified as objects or arrays). Responsive values can currently be achieved using `useBreakpointValue`:

```jsx
const ResponsivelySizedModal = () => {
  const size = useBreakpointValue({base: 'full', lg: 'xl'})
  return (
    <Modal size={size}>
      {/* ... */}
    </Modal>
  )
}
```

```jsx
const ResponsivelySizedModal = () => {
  const size = useBreakpointValue(['full', null, 'xl'])
  return (
    <Modal size={size}>
      {/* ... */}
    </Modal>
  )
}
```

This functionality should instead become native to these props, like most other props in our system:

```jsx
const ResponsivelySizedModal = () => (
  <Modal size={{base: 'full', lg: 'xl'}}>
    {/* ... */}
  </Modal>
)
```

```jsx
const ResponsivelySizedModal = () => (
  <Modal size={['full', null, 'xl']}>
    {/* ... */}
  </Modal>
)
```

**Comment by segunadebayo** [comment]
Thanks for this, Mark.

We can consider this post v1.

**Comment by hutber** [comment]
Thanks for considering this Segun. I would also love to have this feature

**Comment by with-heart** [comment]
We should also support `orientation` as part of this effort. See #2686

**Comment by segunadebayo** [comment]
Thanks for the request! Considering this isn't a bug, I'll add this to our project board so you can track the progress.

Cheers.

**Comment by Philzen** [comment]
What is the status of this – has it been merged or is it in the pipeline for 2.0?

If i understand correctly, this change would make [this caveat](https://chakra-ui.com/docs/styled-system/utility-hooks/use-breakpoint-value) go away: 
> The new `variant` and `size` props don't currently accept responsive values (specified as objects or arrays), but useBreakpointValue is a good way to achieve the same behavior.

Took me quite a while of scratching my head until a friend showed me this sentence, which is somehow well hidden in the docs – it should be mentioned somewhere more upfront as a known limitation.

Also: will this make responsive values in the `defaultProps` section of a `ComponentStyleConfig` work when customizing the default theme?

**Comment by jrolfs** [comment]
Looks like this actually landed (aside from `orientation`?) in #6050. @segunadebayo might want to remove it from [chakra-ui/chakra-ui/projects/6](https://github.com/chakra-ui/chakra-ui/projects/6) or update the status there too?
